packages/flutter/lib/src/cupertino/sliding_segmented_control.dart

```diff
@@ -162,26 +177,21 @@ class _SegmentState<T> extends State<_Segment<T>> with TickerProviderStateMixin<
             child: AnimatedDefaultTextStyle(
               style: DefaultTextStyle.of(context)
                 .style
-                .merge(TextStyle(fontWeight: widget.highlighted ? FontWeight.w500 : FontWeight.normal)),
+                .merge(TextStyle(
+                  fontWeight: widget.highlighted ? FontWeight.w600 : FontWeight.w500,
+                  fontSize: 13,
+                  color: widget.enabled ? null : _kDisabledContentColor,
+                )),
               duration: _kHighlightAnimationDuration,
               curve: Curves.ease,
               child: ScaleTransition(
+                alignment: scaleAlignment,
                 scale: highlightPressScaleAnimation,
                 child: widget.child,
               ),
             ),
           ),
-          // The entire widget will assume the size of this widget, so when a
-          // segment's "highlight" animation plays the size of the parent stays
-          // the same and will always be greater than equal to that of the
-          // visible child (at index 0), to keep the size of the entire
-          // SegmentedControl widget consistent throughout the animation.
-          Offstage(
-            child: DefaultTextStyle.merge(
-              style: const TextStyle(fontWeight: FontWeight.w500),
-              child: widget.child,
-            ),
-          ),
+          widget.child,
```

Comment:
What's the reason for removing the `DefaultTextStyle`? It was there to account for the width change when the segment is highlighted.

Reply:
I tired to keep it but seems only removing both `Offstage` and `DefaultTextStyle` can solve the shaking issue. Keeping DefaultTextStyle still make the overall size shakes a little during thumb switch.

Reply:
It's probably a font difference. If you use a font where the bolder typeface is much wider than the thinner typeface it will probably still expand when you select a wider segment.

Reply:
The reason it expands was some segments can become a lot wider when "highlighted" is true (as the typeface used becomes bold). So child 1 basically always stay in the "highlighted" state to make sure the segment doesn't change width when child 0 becomes highlighted.

Since you updated the font weight and size for child 0 you probably want to update the font weight and size for child 1 too.